MazerPlatformer/GameWorld.cs

```diff
@@ -126,15 +128,16 @@ public void SaveLevel()
         /// Listen for collision events
         /// Listen for scoring, special moves, power-ups etc
         /// </summary>
-        public void Initialize()
+        public Either<IFailure, Unit> Initialize()
         {
             // Hook up the Player events to the external world ie game UI
-            _level.Player.OnStateChanged += state => OnPlayerStateChanged?.Invoke(state); // want to know when the player's state changes
-            _level.Player.OnDirectionChanged += direction => OnPlayerDirectionChanged?.Invoke(direction); // want to know when the player's direction changes
-            _level.Player.OnCollisionDirectionChanged += direction => OnPlayerCollisionDirectionChanged?.Invoke(direction); // want to know when player collides
-            _level.Player.OnGameObjectComponentChanged += (thisObject, name, type, oldValue, newValue) => OnPlayerComponentChanged?.Invoke(thisObject, name, type, oldValue, newValue); // want to know when the player's components change
-            _level.Player.OnCollision += PlayerOnOnCollision;
-            _level.Player.OnPlayerSpotted += (sender, args) => _level.PlayPlayerSpottedSound(); 
+            Level.Player.OnStateChanged += state => Ensure(()=> OnPlayerStateChanged?.Invoke(state)); // want to know when the player's state changes
```

Comment:
I dont like all these ensures - they really hide and muddy the true code intention